spec/coder_spec.lua

```diff
@@ -298,315 +298,126 @@ describe("Titan coder /", function()
-            end)
-
-            it("integer (+)", function()
-                run_binop_test("add_int", "+", "math.maxinteger", "1")
-                run_binop_test("add_int", "+", "math.mininteger", "-1")
-            end)
-
-            it("integer (//)", function()
-                run_binop_test("intdiv_int", "//", " 10", " 3")
-                run_binop_test("intdiv_int", "//", " 10", "-3")
-                run_binop_test("intdiv_int", "//", "-10", " 3")
-                run_binop_test("intdiv_int", "//", "-10", "-3")
-                run_binop_test("intdiv_int", "//", "math.mininteger", "-1")
-            end)
-
-            it("integer (%)", function()
-                run_binop_test("mod_int", "%", " 10", " 3")
-                run_binop_test("mod_int", "%", " 10", "-3")
-                run_binop_test("mod_int", "%", "-10", " 3")
-                run_binop_test("mod_int", "%", "-10", "-3")
-                run_binop_test("mod_int", "%", "math.mininteger", "-1")
-            end)
+            lua_tests[name] = util.render([[
+                local test_operators = require "test_operators"
+                test_operators.check_binop(
+                    $op_str,
+                    function(x, y) return (x $op y) end,
+                    test.${name},
+                    $typ1_str,
+                    $typ2_str
+                )
+            ]],{
+                name = name,
+                op = op,
+                op_str = string.format("%q", op),
+                typ1_str = string.format("%q", typ1),
+                typ2_str = string.format("%q", typ2),
+            })
+        end
 
-            it("right shift (>>)", function()
-                run_binop_test("rshift", ">>", "0xdead", " 1")
-                run_binop_test("rshift", ">>", "0xdead", "-1")
-                run_binop_test("rshift", ">>", "0xdead", " 100")
-                run_binop_test("rshift", ">>", "0xdead", "-100")
-                run_binop_test("rshift", ">>", "0xdead", "math.maxinteger")
-                run_binop_test("rshift", ">>", "0xdead", "math.mininteger")
-            end)
+        local function optest(name)
+            run_test(lua_tests[name])
+        end
 
-            it("left shift (<<)", function()
-                run_binop_test("lshift", "<<", "0xdead", " 1")
-                run_binop_test("lshift", "<<", "0xdead", "-1")
-                run_binop_test("lshift", "<<", "0xdead", " 100")
-                run_binop_test("lshift", "<<", "0xdead", "-100")
-                run_binop_test("lshift", "<<", "0xdead", "math.maxinteger")
-                run_binop_test("lshift", "<<", "0xdead", "math.mininteger")
-            end)
-        end)
+        setup_unop("neg_int", "-",   "integer", "integer")
+        setup_unop("bnot",    "~",   "integer", "integer")
+        setup_unop("not_bool","not", "boolean", "boolean")
+
+        setup_binop("add_int"       , "+", "integer", "integer", "integer")
```

Comment:
In the previous block the comma was aligned with argument in the left, here it is aligned in the right. We should be consistent.
Also, we could align the other arguments.